src/test/java/org/manifold/compiler/front/TestExpressionGraph.java

```diff
@@ -1,5 +1,22 @@
 package org.manifold.compiler.front;
 
+import org.junit.Before;
+import org.junit.Test;
+import org.manifold.compiler.BooleanValue;
+
 public class TestExpressionGraph {
+  ExpressionGraph expressionGraph;
+
+  @Before
+  public void setup() {
+    expressionGraph = new ExpressionGraph();
+  }
 
+  @Test(expected = IllegalArgumentException.class)
+  public void testAddEdge_verticesNotInGraph_exceptionThrown() {
+    ExpressionVertex from = new ConstantValueVertex(expressionGraph, BooleanValue.getInstance(true));
+    ExpressionVertex to = new ConstantValueVertex(expressionGraph, BooleanValue.getInstance(false));
+    ExpressionEdge edgeOfTomCruise = new ExpressionEdge(from, to);
+    expressionGraph.addEdge(edgeOfTomCruise);
+  }
```

Comment:
It seems like there are a couple more cases in your code. To get complete branch coverage you should test

1. When the target is null
2. When the target is not in the graph
2. When the source is not in the graph